oecophylla/cli/launch.py

```diff
@@ -0,0 +1,190 @@
+    _create_dir(output_dir)
+
+    # SAMPLE SHEET
+    # If a manifest is not specified, the files containing forward and reverse reads
+    # will be automatically identified using regex.
+    if sample_sheet:
+        _sheet = read_sample_sheet(sample_sheet)
+        sample_dict = extract_samples_from_sample_sheet(_sheet, input_dir)
+    else:
+        sample_dict = extract_sample_paths(input_dir)
+
+    # PARAMS
+    with open(params, 'r') as f:
+        params_dict = yaml.load(f)
+
+    # ENVS
+    with open(envs, 'r') as f:
+        envs_dict = yaml.load(f)
+
+    # CONFIG
+    # merge PARAMS, SAMPLE_DICT, ENVS
+    config_dict = {}
+    config_dict['samples'] = sample_dict
+    config_dict['params'] = params_dict
+    config_dict['envs'] = envs_dict
+
+    config_yaml = yaml.dump(config_dict, default_flow_style=False)
+    config_fp = '%s/%s' % (local_scratch, 'config.yaml')
+    with open(config_fp, 'w') as f:
+        f.write(config_yaml)
+
+    # LOGS
+    if log_dir:
+        _create_dir(log_dir)
+    else:
+        os.makedirs('%s/%s' % (output_dir, 'cluster_logs'))
+
+    # CLUSTER SETUP
+    with open(cluster_params) as _file:
+        _cluster_config = yaml.load(_file)
+    # for now, everything under `extra` should be explicit freetext,
+    # e.g. --my-argument=value
+    cluster_freetext = _cluster_config['extra']
+    if workflow_type == 'torque':
+        cluster_setup = "qsub -e {cluster.error} -o {cluster.output} \
+                         -m {cluster.email} \
+                         -l nodes=1:ppn={cluster.nodes} \
+                         -l mem={cluster.memory} \
+                         -l walltime={cluster.time} %s" % cluster_freetext
+    elif workflow_type == 'slurm':
+        cluster_setup = "srun -e {cluster.error} -o {cluster.output} \
+                         -mail-user={cluster.email} \
+                         -n {cluster.nodes} \
+                         --mem={cluster.memory} \
+                         --time={cluster.time} %s" % cluster_freetext
+    elif workflow_type == 'local':
+        if not cluster_params:
+            cluster['cores'] = 4
+            cluster['nodes'] = 1
+            cluster['local_cores'] = 1
+        cluster_setup = None
+    else:
+        raise ValueError('Incorrect run-location specified in launch script.')
+
+    snakemake.snakemake(snakefile,
+                        cores=cluster['cores'],
+                        nodes=cluster['nodes'],
+                        local_cores=cluster['local_cores'],
+                        cluster=cluster_setup,
+                        cluster_config=cluster_params,
+                        workdir="$@",
+                        forceall=force,
+                        config=config_fp,
+                        latency_wait=latency_wait)
+
+
+@run.command()
+@click.option('--config-file', default=None,
+              help='Configuration file defining specific installations of all'
+                   ' modules.')
+def install():
```

Comment:
`config_file` is not a parameter to the function

Reply:
we're not going focus on the install script - so we'll postpone to a future date